### PR TITLE
Add documentation for internal JWT auth

### PR DESCRIPTION
**Pull request by dain**

### CODE REVIEW

**Comment by mosabua** [comment]
Saw your mention late @dain  ... docs look good.

File: presto-docs/src/main/sphinx/security/internal-communication.rst [hunk 1]
```diff
@@ -5,6 +5,23 @@ Secure Internal Communication
 The Presto cluster can be configured to use secured communication. Communication
 between Presto nodes can be secured with SSL/TLS.
 
+Internal Authentication
+-----------------------
+
+Requests between Presto nodes are authenticated using a shared secret.  The shared
```
Comment (mosabua) [inline presto-docs/src/main/sphinx/security/internal-communication.rst]: Off topic question ... when you talk about nodes you mean all controllers and workers in all clusters right? Or just all workers..

Comment (dain) [reply presto-docs/src/main/sphinx/security/internal-communication.rst]: "all nodes" being every Presto process... coordinators and workers

Comment (mosabua) [reply presto-docs/src/main/sphinx/security/internal-communication.rst]: Thanks for explanation.

**Pull request merged**